scripts/trace_flowcell.py

```diff
@@ -2,117 +2,145 @@
 import requests
 import json
 import sys
+import subprocess
 import yaml
 import argparse
 from datetime import datetime
 
+class get_stuff:
+    def __init__(self, api_base_url, access_headers, req_verify, tag):
+        self.api_base_url = api_base_url
+        self.headers = access_headers
+        self.verify = req_verify
+        self.tag = tag
+    
+    def get_traces_for_tag(self):
+        traces_response = requests.get(
+            "{}/{}/?{}={}".format(self.api_base_url, "traces", "trace_tag", self.tag),
+            verify = self.verify,
+            headers = self.headers)
+        if not traces_response.ok:
+            raise Exception("Response not OK, got: " + traces_response.text)
+        self.traces = json.loads(traces_response.text)
+        return json.loads(traces_response.text)
+    
+    def get_executions_from_traces(self):
+        for trace in self.traces:
+            trace_id = trace["id"]
+            trace_info_response = requests.get(
+                "{}/{}/{}".format(self.api_base_url, "traces", trace_id),
+                verify = self.verify,
+                headers = self.headers)
+            trace_info = json.loads(trace_info_response.text)
+            action_executions = map(lambda x: x["object_id"], trace_info["action_executions"])
+            for execution in action_executions:
+                yield execution
+
+    def get_actions_from_executions(self, executions):
+        for execution_id in executions:
+            execution_info_response = requests.get(
+                "{}/{}/{}".format(self.api_base_url, "executions", execution_id),
+                verify = self.verify,
+                headers = self.headers)
+            execution_info = json.loads(execution_info_response.text)
+            yield execution_info
+    
+    def filter_actions_by_name(self, actions, name):
+        self.filtered_actions = (action for action in actions if name in action["action"]["name"])
+
+    def sort_actions_by_timestamp(self):
+        def get_start_time(action):
+            start_time = datetime.strptime(action['start_timestamp'].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+            return start_time
+        sort_actions = sorted(self.filtered_actions, key = get_start_time)
+        return sort_actions
+
+
+class runfolders:
+    def __init__(self):
+
+        with open(args.config, 'r') as ymlfile:
+            cfg = yaml.load(ymlfile)
+        hosts = cfg['runfolder_svc_urls']
+
+        self.hitcount = 0
+        self.outstring = "\n\tstatus\trunfolder_link\n"
```

Comment:
you don't need to initialize the outstring here and it doesn't need to be a instance variable (e.g. if you call `pick_runfolder` twice you would get unexpected output)